components/eam/src/chemistry/modal_aero/modal_aero_amicphys_subareas.F90

```diff
@@ -0,0 +1,649 @@
+  integer :: icnst    ! consitituent index 
+
+  real(wp) :: qgcm_intrst  ! grid cell mean of interstitial aerosol mixing ratio of a single mode
+  real(wp) :: qgcm_cldbrn  ! grid cell mean of cloud-borne  aerosol mixing ratio of a single mode
+
+  real(wp) :: factor_clea  ! partitioning factor for clear  subarea [unitless]
+  real(wp) :: factor_cldy  ! partitioning factor for cloudy subarea [unitless]
+
+  do imode = 1, ntot_amode
+
+     ! calculcate partitioning factors
+
+     qgcm_intrst = 0.0_wp
+     qgcm_cldbrn = 0.0_wp
+
+     do ispec = 1, nspec_amode(imode)
+        qgcm_intrst = qgcm_intrst + qgcm( lmassptr_amode(ispec,imode) - loffset )
+        qgcm_cldbrn = qgcm_cldbrn + qqcwgcm( lmassptrcw_amode(ispec,imode) - loffset )
+     end do
+
+     call get_partition_factors( qgcm_intrst, qgcm_cldbrn, fcldy, fclea, &
+                                 factor_clea, factor_cldy )
+
+     ! apply partitioning factors
+
+     do ispec = 1, nspec_amode(imode)
+
+        icnst = lmassptr_amode(ispec,imode) - loffset
+
+        qsubx(icnst,jclea) = qgcmx(icnst)*factor_clea
+        qsubx(icnst,jcldy) = qgcmx(icnst)*factor_cldy
+
+     end do ! ispec 
+  end do ! imode
+
+
+end subroutine set_subarea_qmass_for_intrst_aerosols
+
+subroutine get_partition_factors(  qgcm_intrst, qgcm_cldbrn, fcldy, fclea, &! in
+                                   factor_clea, factor_cldy                )! out
+!------------------------------------------------------------------------------------
+! Purpose: Calculate the partitioning factors for distributing interstitial aerosol
+!          mixing ratios to cloudy and clear subareas in a grid box.
+!          The partitioning factors depend on the grid cell mean mixing ratios of
+!          both interstitial and cloud-borne aerosols.
+!------------------------------------------------------------------------------------
+
+  real(wp), intent(in)  ::  qgcm_intrst  ! grid cell mean interstitial aerosol mixing ratio
+  real(wp), intent(in)  ::  qgcm_cldbrn  ! grid cell mean cloud-borne aerosol mixing ratio
+
+  real(wp), intent(in)  ::  fcldy        ! cloudy fraction of the grid cell [unitless]
+  real(wp), intent(in)  ::  fclea        ! clear  fraction of the grid cell [unitless]
+
+  real(wp), intent(out) ::  factor_clea  ! partitioning factor for clear  subarea
+  real(wp), intent(out) ::  factor_cldy  ! partitioning factor for cloudy subarea
+
+  real(wp) :: tmp_q_intrst_clea, tmp_q_intrst_cldy
+  real(wp) :: tmp_q_cldbrn_cldy
+  real(wp) :: clea2gcm_ratio
+
+  real(wp),parameter :: eps = 1.e-35_wp
+
+  ! Calculate subarea-mean mixing ratios
+
+  tmp_q_cldbrn_cldy = qgcm_cldbrn/fcldy                                              ! cloud-borne,  cloudy subarea
+  tmp_q_intrst_cldy = max( 0.0_wp, ((qgcm_intrst+qgcm_cldbrn) - tmp_q_cldbrn_cldy) ) ! interstitial, cloudy subarea
+
+  tmp_q_intrst_clea = (qgcm_intrst - fcldy*tmp_q_intrst_cldy)/fclea                  ! interstitial, clear  subarea
```

Comment:
There is a similar assumption about `fclea` being positive here.

Reply:
Same as above. Added the following comment:

```
  ! Note that this subroutine is designed for partially cloudy grid cells,
  ! hence both fclea and fcldy are assumed to be nonzero.
```